corehq/apps/reports/tests/test_tableau_api.py

```diff
@@ -1,34 +1,328 @@
-from django.test import SimpleTestCase
-from corehq.apps.reports.models import TableauConnectedApp, TableauServer
+import requests
+from unittest import mock
+from django.test import TestCase
+from corehq.apps.reports.exceptions import TableauAPIError
+from corehq.apps.reports.models import TableauAPISession, TableauConnectedApp, TableauServer
 
 
-class TestTableauConnectedApp(SimpleTestCase):
+class FakeTableauInstance(mock.MagicMock):
+
+    def __init__(self):
+        super(FakeTableauInstance, self).__init__()
+        self.groups = {'group1': '1a2b3', 'group2': 'c4d5e'}
+        self.users = {'angie@dimagi.com': 'zx8cv', 'jeff@company.com': 'uip12', 'steve@company.com': 'ty78ui'}
+        self.group_names = list(self.groups.keys())
+        self.group_ids = list(self.groups.values())
+        self.users_names = list(self.users.keys())
+        self.user_ids = list(self.users.values())
+
+    def _create_response(self, response_text):
+        mock_response = requests.Response()
+        mock_response.status_code = 200
+        mock_response._content = response_text.encode('utf-8')  # Sets the return value for response.text
+        return mock_response
+
+    def API_version_response(self):
+        return self._create_response("""
+        {
+            "serverInfo": {
+                "productVersion": {
+                    "value": "2022.1.5",
+                    "build": "20221.22.0823.1450"
+                },
+                "restApiVersion": "3.15"
+            }
+        }
+        """)
+
+    def sign_in_response(self):
+        return self._create_response("""
+        {
+            "credentials": {
+                "site": {
+                    "id": "asdfasdf-1234-5678-90as-0s0s0s0s0s0s",
+                    "contentUrl": "Test"
+                },
+                "user": {
+                    "id": "123d4567-8901-2e34-5678-9t9t9t9t9t9t"
+                },
+                "token": "aaaaaaaaaaaaaaaaaaaaaa22222222222222222222222",
+                "estimatedTimeToExpiration": "364:00:42"
+            }
+        }
+        """)
+
+    def query_groups_response(self, **kwargs):
+        if 'group_name' in kwargs:
+            return self._create_response("""
+                {
+                    "groups": {
+                        "group": [
```

Comment:
is this really what the JSON looks like? I can't find any docs on the site that have example responses in JSON, only XML.

I expected it to look like `{"groups": [{...}, {...}] }`